### PR TITLE
Chameleon paginator

### LINKED ISSUE #61: New Element: chameleon-paginator

### PR DESCRIPTION
**Pull request by ryuhhnn**
closes #61

### CODE REVIEW

File: packages/paginator/src/chameleon-paginator.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  property,
+  html,
+  svg,
+  SVGTemplateResult
+} from "lit-element";
+import base from "@chameleon-ds/theme/base";
+import style from "@chameleon-ds/theme/base/paginator";
+import "@chameleon-ds/button";
+
+@customElement("chameleon-paginator")
```
Comment (jbforth4) [inline packages/paginator/src/chameleon-paginator.ts]: This sure looks an awful lot like a Java/Spring class 🤣

**Comment by mandymichel** [comment]
If you change the values of the Total Items and Page Size, it will initially sometimes render NaN instead of the expected numbers. Also, when there are less than 8 pages, the first page doesn't have the circle marker on it (maybe in other scenarios, too). I think the numbers might be a little too tall in comparison with the end arrows, but I am having trouble finding exact height with dev tools for some reason.

**Comment by ryuhhnn** [comment]
Good catch @mandymichel!

> If you change the values of the Total Items and Page Size, it will initially sometimes render NaN instead of the expected numbers.

I couldn't reproduce this consistently, but I definitely could reproduce it. Added a check that seems to have fixed this

> Also, when there are less than 8 pages, the first page doesn't have the circle marker on it (maybe in other scenarios, too).

Haven't been able to reproduce this one

> I think the numbers might be a little too tall in comparison with the end arrows, but I am having trouble finding exact height with dev tools for some reason.

The measurements we have in our internal Sketch documents are a little off, but I tried to match them as best as I could. I'm definitely going to circle around with UX and create a new ticket to improve these styles based on their feedback

**Comment by ZempTime** [comment]
Potentially out of the purview of this component in terms of properties, but not in terms of what it'd be working with - [Prisma's Pagination Approach](https://www.prisma.io/docs/prisma-client/basic-data-access/reading-data-TYPESCRIPT-rsc3/#pagination). Cursor-based.

Example:
```
// data gotten
this.users = users.where({first: 10});
```

Screen Shot 2019-10-18 at 11 09 52 AM

Need to think through how pagination actually plays out more

**Comment by mandymichel** [comment]
I'm still seeing the same problems as before. Not sure if my local setup is picking up the new change- if I need to rebuild or something. I notice that a debugger in your new if statement doesn't work but that may be unrelated.

**Comment by kewalter** [comment]
I'm seeing the same NaN issue as @mandymichel. When changing the numbers, it will occasionally display NaN.  

Also, and maybe this is an improvement later, we should be hiding the prev/next arrows when we're on the first/last pages or if there's only a single page of items.

**Pull request merged**